### PR TITLE
docs: add page on pruning a geth node

### PR DESCRIPTION
**Pull request by jmcook1186**
Adds a page with brief instructions for pruning a geth node.
Also intended for use on new site.

### CODE REVIEW

File: docs/_interface/pruning.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+3) Geth is at least `v1.10` ideally > `v1.10.3`
+4) Geth is fully sync'd
+5) Geth has finished creating a snapshot that is at least 128 blocks old. This is true when 
+   "state snapshot generation" is no longer reported in the logs.
+
+With these rules satisfied, Geth's database can be pruned.
+
+## How pruning works
+
+Pruning uses snapshots of the state database as an acceleration structure to determine which
```
Comment (rjl493456442) [inline docs/_interface/pruning.md:34]: Pruning uses snapshots of the state database as an indicator to determine nodes ...

File: docs/_interface/pruning.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+   "state snapshot generation" is no longer reported in the logs.
+
+With these rules satisfied, Geth's database can be pruned.
+
+## How pruning works
+
+Pruning uses snapshots of the state database as an acceleration structure to determine which 
+nodes in the state trie can be kept and which ones are stale and can be discarded. Geth 
+rebuilds the state trie from a stored snapshot that is older than 128 blocks, discarding any 
+data that isn't part of the snapshot window (the sequence of blocks between the snapshot and head).
```
Comment (rjl493456442) [inline docs/_interface/pruning.md:37]: discarding data that isn't part of the target state trie or genesis state

File: docs/_interface/pruning.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+## How pruning works
+
+Pruning uses snapshots of the state database as an acceleration structure to determine which 
+nodes in the state trie can be kept and which ones are stale and can be discarded. Geth 
+rebuilds the state trie from a stored snapshot that is older than 128 blocks, discarding any 
+data that isn't part of the snapshot window (the sequence of blocks between the snapshot and head). 
+
+Geth prunes the database in three stages:
+
+1) Iterating state snapshot: Geth iterates through states, starting at the most recent snapshot,
```
Comment (rjl493456442) [inline docs/_interface/pruning.md:41]: 1) Iterating state snapshot: Geth iterates the bottom-most snapshot layer and construct a bloom filter set for identifying the target trie nodes.

File: docs/_interface/pruning.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+Pruning uses snapshots of the state database as an acceleration structure to determine which 
+nodes in the state trie can be kept and which ones are stale and can be discarded. Geth 
+rebuilds the state trie from a stored snapshot that is older than 128 blocks, discarding any 
+data that isn't part of the snapshot window (the sequence of blocks between the snapshot and head). 
+
+Geth prunes the database in three stages:
+
+1) Iterating state snapshot: Geth iterates through states, starting at the most recent snapshot, 
+   to identify stale trie nodes.
+2) Pruning state data: Geth deletes stale trie nodes from the database.
```
Comment (rjl493456442) [inline docs/_interface/pruning.md:43]: Geth deletes stale trie nodes from the database which are not in the bloom filter set.

File: docs/_interface/pruning.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+5) Geth has finished creating a snapshot that is at least 128 blocks old. This is true when 
+   "state snapshot generation" is no longer reported in the logs.
+
+With these rules satisfied, Geth's database can be pruned.
+
+## How pruning works
+
+Pruning uses snapshots of the state database as an acceleration structure to determine which 
+nodes in the state trie can be kept and which ones are stale and can be discarded. Geth 
+rebuilds the state trie from a stored snapshot that is older than 128 blocks, discarding any
```
Comment (rjl493456442) [inline docs/_interface/pruning.md:36]: Geth identifies the target state trie based on a stored snapshot layer which has at least 128 block confirmations on top(for surviving reorgs)

File: docs/_interface/pruning.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+sudo -u <user> geth --datadir <path> snapshot prune-state # wait for pruning to finish
+sudo systemctl start geth # restart geth
+```
+
+The pruning could take 4-5 hours to complete. Once finished, restart Geth.
+
+
+## Troubleshooting
+
+Messages about "state snapshot generation" indicate that a snapshot does not actually exist.
```
Comment (rjl493456442) [inline docs/_interface/pruning.md:74]: a snapshot is not fully generated.

**Review by rjl493456442 — COMMENTED** [review]
Thanks for the doc!

Could please add another section or notification that: offline pruning is only for the hash-based state scheme, and we will have path-based state scheme be delivered in the following months which enables the pruning by default. And we will deprecate this tool once hash-based state scheme is not supported in the future.

File: docs/_interface/pruning.md (hunk 1 above)
Comment (jmcook1186) [reply docs/_interface/pruning.md:34]:
```suggestion
Pruning uses snapshots of the state database as an indicator to determine which 
```

File: docs/_interface/pruning.md (hunk 2 above)
Comment (jmcook1186) [reply docs/_interface/pruning.md:37]:
```suggestion
data that isn't part of the target state trie or genesis state. 
```

File: docs/_interface/pruning.md (hunk 3 above)
Comment (jmcook1186) [reply docs/_interface/pruning.md:41]:
```suggestion
1) Iterating state snapshot: Geth iterates the bottom-most snapshot layer and constructs a bloom filter set for identifying the target trie nodes.
```

File: docs/_interface/pruning.md (hunk 4 above)
Comment (jmcook1186) [reply docs/_interface/pruning.md:43]:
```suggestion
2) Pruning state data: Geth deletes stale trie nodes from the database which are not in the bloom filter set.
```

File: docs/_interface/pruning.md (hunk 6 above)
Comment (jmcook1186) [reply docs/_interface/pruning.md:74]:
```suggestion
Messages about "state snapshot generation" indicate that a snapshot is not fully generated. 
```

File: docs/_interface/pruning.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+Pruning uses snapshots of the state database as an indicator to determine which 
+nodes in the state trie can be kept and which ones are stale and can be discarded. Geth 
+identifies the target state trie based on a stored snapshot layer which has at least 128 block confirmations on top(for surviving reorgs)
+data that isn't part of the target state trie or genesis state. 
+
+Geth prunes the database in three stages:
+
+1) Iterating state snapshot: Geth iterates the bottom-most snapshot layer and constructs a bloom filter set for identifying the target trie nodes.
+   to identify stale trie nodes.
```
Comment (jmcook1186) [inline docs/_interface/pruning.md:42]:
```suggestion
```

**Comment by rjl493456442** [comment]
The content lgtm. But `{% include note.html` is not the markdown grammar, is it OK? cc @fjl

**Pull request merged**